src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs

```diff
@@ -283,44 +246,20 @@ private string GetObjectIdByEmail(string email)
             if (DefaultProfile.DefaultContext.Environment.OnPremise || string.IsNullOrWhiteSpace(email)) return null;
 
             string objId = null;
-// TODO: Remove IfDef
-#if NETSTANDARD
             var users = ActiveDirectoryClient.FilterUsers(new ADObjectFilterOptions { Mail = email });
             if (users != null)
             {
                 ThrowIfMultipleObjectIds(users, email);
                 var user = users.FirstOrDefault();
                 objId = user?.Id.ToString();
             }
-#else
-            var users = ActiveDirectoryClient.Users.Where(FilterByEmail(email)).ExecuteAsync().GetAwaiter().GetResult().CurrentPage;
```

Comment:
This code finds emails addresses that are in either the email property, or the otherMails attribute of AAD user objects. The that you are keeping above looks like it only checks the email attribute and does not include the otherMails attribute.  That will change the behavior of the cmdlet.